feat: Separate frontend and backend into distinct projects

AlexMakesC0de opened the pull request:
- Created PassManGUI as standalone Blazor Web App
- Removed Blazor components from PassManAPI (now REST API only)
- Added CORS configuration to PassManAPI for frontend access
- Added HttpClient configuration to PassManGUI
- Updated docker-compose.yml with PassManGUI service
- Created Dockerfile for PassManGUI

AlexMakesC0de committed:
feat: Separate frontend and backend into distinct projects

- Created PassManGUI as standalone Blazor Web App
- Removed Blazor components from PassManAPI (now REST API only)
- Added CORS configuration to PassManAPI for frontend access
- Added HttpClient configuration to PassManGUI
- Updated docker-compose.yml with PassManGUI service
- Created Dockerfile for PassManGUI

CosminBuzicUni reviewed (dismissed):
Doesn't build. Both the PassManAPI, PassManGUI dockerfiles are targeting asp.net versions that don't support some features like  createScopeForStatusCodePages. This will only run locally if the installed sdk supports it, but it will not run containerized. SDK version target must be changed

AlexMakesC0de committed:
feat: Add Docker hot reload support and fix environment config

- Add dockerfile.dev for PassManAPI and PassManGUI with dotnet watch
- Update docker-compose.yml to use volume mounts for hot reload
- Override connection string via environment variable for Docker
- Update CORS to allow requests from Docker and local dev ports
- Remove redundant appsettings.Docker.json files

pop9459 committed:
Merge branch 'develop' into feature/separate-frontend-backend

pop9459 commented:
works after cleaning old build files
`dotnet restore`
`dotnet clean`
`rm -rf bin obj`
